src/libPersistence/BlockStorage.cpp

```diff
@@ -170,6 +195,8 @@ bool BlockStorage::GetRangeMicroBlocks(const uint64_t lowEpochNum,
                                        list<MicroBlockSharedPtr>& blocks) {
   LOG_MARKER();
 
+  shared_lock<shared_timed_mutex> g(m_mutexMicroBlock);
+
   leveldb::Iterator* it =
       m_microBlockDB->GetDB()->NewIterator(leveldb::ReadOptions());
   for (it->SeekToFirst(); it->Valid(); it->Next()) {
```

Comment:
Maybe let `m_mutexMicroBlock` to protect `leveldb::Iterator* it = m_microBlockDB->GetDB()->NewIterator(leveldb::ReadOptions());` only

Reply:
cannot do that...the iterator is keep changing in the for loop